File: src/main/java/com/DavidM1A2/AfraidOfTheDark/common/constants/ModEntities.java
package com.DavidM1A2.afraidofthedark.common.constants;

import com.DavidM1A2.afraidofthedark.common.entity.bolt.EntityIronBolt;
import com.DavidM1A2.afraidofthedark.common.entity.bolt.EntitySilverBolt;
import com.DavidM1A2.afraidofthedark.common.entity.bolt.EntityWoodenBolt;
import com.DavidM1A2.afraidofthedark.common.entity.enchantedSkeleton.EntityEnchantedSkeleton;
import com.DavidM1A2.afraidofthedark.common.entity.werewolf.EntityWerewolf;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.fml.common.registry.EntityEntry;
import net.minecraftforge.fml.common.registry.EntityEntryBuilder;

/**
 * A static class containing all of our entity references for us
 */
public class ModEntities
{
	// Various entity IDs
	public static final int WEREWOLF_ID = 0;
	public static final int IRON_BOLT_ID = 1;
	public static final int SILVER_BOLT_ID = 2;
	public static final int WOODEN_BOLT_ID = 3;
	public static final int DEEE_SYFT_ID = 4;
	public static final int IGNEOUS_BOLT_ID = 5;
	public static final int STAR_METAL_BOLT_ID = 6;
	public static final int ENCHANTED_SKELETON_ID = 7;
	public static final int ENARIA_ID = 8;
	public static final int SPLINTER_DRONE_ID = 9;
	public static final int SPLINTER_DRONE_PROJECTILE_ID = 10;
	public static final int SPELL_PROJECTILE_ID = 11;
	public static final int SPELL_PROJECTILE_DIVE_ID = 12;
	public static final int SPELL_MYSELF_ID = 13;
	public static final int SPELL_AOE_ID = 14;
	public static final int GHASTLY_ENARIA_ID = 15;
	public static final int ARTWORK_ID = 16;
	public static final int SPELL_EXTRA_EFFECTS_ID = 17;

	// All mod entity static fields
	public static final EntityEntry ENCHANTED_SKELETON = EntityEntryBuilder.create()
			.egg(0x996600, 0xe69900)
			.entity(EntityEnchantedSkeleton.class)
			.id(new ResourceLocation(Constants.MOD_ID, "enchanted_skeleton"), ENCHANTED_SKELETON_ID)
			.name("enchanted_skeleton")
			.tracker(50, 1, true)
			.build();
	public static final EntityEntry WEREWOLF = EntityEntryBuilder.create()
			.egg(0x3B170B, 0x181907)
			.entity(EntityWerewolf.class)
			.id(new ResourceLocation(Constants.MOD_ID, "werewolf"), WEREWOLF_ID)
			.name("werewolf")
			.tracker(50, 1, true)
			.build();

	// 5 bolt entities
	public static final EntityEntry WOODEN_BOLT = EntityEntryBuilder.create()
			.entity(EntityWoodenBolt.class)
			.id(new ResourceLocation(Constants.MOD_ID, "wooden_bolt"), WOODEN_BOLT_ID)
			.name("wooden_bolt")
			.tracker(50, 1, true)
			.build();
	public static final EntityEntry IRON_BOLT = EntityEntryBuilder.create()
			.entity(EntityIronBolt.class)
			.id(new ResourceLocation(Constants.MOD_ID, "iron_bolt"), IRON_BOLT_ID)
			.name("iron_bolt")
			.tracker(50, 1, true)
			.build();
	public static final EntityEntry SILVER_BOLT = EntityEntryBuilder.create()
			.entity(EntitySilverBolt.class)
			.id(new ResourceLocation(Constants.MOD_ID, "silver_bolt"), SILVER_BOLT_ID)
			.name("silver_bolt")
			.tracker(50, 1, true)
			.build();

	// An array containing a list of entities that AOTD adds
	public static EntityEntry[] ENTITY_LIST = new EntityEntry[]
	{
		ENCHANTED_SKELETON,
		WEREWOLF,
		WOODEN_BOLT,
		IRON_BOLT,
		SILVER_BOLT
	};
}

File: src/main/java/com/davidm1a2/afraidofthedark/common/constants/ModEntities.kt
package com.davidm1a2.afraidofthedark.common.constants

import com.davidm1a2.afraidofthedark.common.entity.bolt.*
import com.davidm1a2.afraidofthedark.common.entity.enaria.EntityEnaria
import com.davidm1a2.afraidofthedark.common.entity.enaria.EntityGhastlyEnaria
import com.davidm1a2.afraidofthedark.common.entity.enchantedFrog.EntityEnchantedFrog
import com.davidm1a2.afraidofthedark.common.entity.enchantedSkeleton.EntityEnchantedSkeleton
import com.davidm1a2.afraidofthedark.common.entity.spell.projectile.EntitySpellProjectile
import com.davidm1a2.afraidofthedark.common.entity.splinterDrone.EntitySplinterDrone
import com.davidm1a2.afraidofthedark.common.entity.splinterDrone.EntitySplinterDroneProjectile
import com.davidm1a2.afraidofthedark.common.entity.werewolf.EntityWerewolf
import net.minecraft.entity.EntityType

/**
 * A static class containing all of our entity references for us
 */
object ModEntities {
    // All mod entity static fields
    val ENCHANTED_SKELETON = EntityType.register(
        "${Constants.MOD_ID}:enchanted_skeleton",
        EntityType.Builder.create(EntityEnchantedSkeleton::class.java) { EntityEnchantedSkeleton(it) }
            .tracker(50, 1, true)
    )
    //.egg(0x996600, 0xe69900)

    val WEREWOLF = EntityType.register(
        "${Constants.MOD_ID}:werewolf",
        EntityType.Builder.create(EntityWerewolf::class.java) { EntityWerewolf(it) }
            .tracker(50, 1, true)
    )
    //.egg(0x3B170B, 0x181907)
    //.spawn(EnumCreatureType.MONSTER, 25, 1, 4, ModBiomes.EERIE_FOREST) // Weight = 100 is for skeletons, use 1/4 of that

    val GHASTLY_ENARIA = EntityType.register(
        "${Constants.MOD_ID}:ghastly_enaria",
        EntityType.Builder.create(EntityGhastlyEnaria::class.java) { EntityGhastlyEnaria(it) }
            .tracker(ModServerConfiguration.blocksBetweenIslands / 2, 1, true)
    )

    val SPLINTER_DRONE = EntityType.register(
        "${Constants.MOD_ID}:splinter_drone",
        EntityType.Builder.create(EntitySplinterDrone::class.java) { EntitySplinterDrone(it) }
            .tracker(50, 1, true)
    )
    //.egg(0xcc6600, 0x63300)

    val SPLINTER_DRONE_PROJECTILE = EntityType.register(
        "${Constants.MOD_ID}:splinter_drone_projectile",
        EntityType.Builder.create(EntitySplinterDroneProjectile::class.java) { EntitySplinterDroneProjectile(it) }
            .tracker(50, 1, true)
    )

    val ENARIA = EntityType.register(
        "${Constants.MOD_ID}:enaria",
        EntityType.Builder.create(EntityEnaria::class.java) { EntityEnaria(it) }
            .tracker(50, 1, true)
    )

    val ENCHANTED_FROG = EntityType.register(
        "${Constants.MOD_ID}:enchanted_frog",
        EntityType.Builder.create(EntityEnchantedFrog::class.java) { EntityEnchantedFrog(it) }
            .tracker(50, 1, true)
    )
    // .egg(0x92029c, 0xd4028e)

    // Spell entities
    val SPELL_PROJECTILE = EntityType.register(
        "${Constants.MOD_ID}:spell_projectile",
        EntityType.Builder.create(EntitySpellProjectile::class.java) { EntitySpellProjectile(it) }
            .tracker(50, 1, true)
    )

    // 5 bolt entities
    val WOODEN_BOLT = EntityType.register(
        "${Constants.MOD_ID}:wooden_bolt",
        EntityType.Builder.create(EntityWoodenBolt::class.java) { EntityWoodenBolt(it) }
            .tracker(50, 1, true)
    )
    val IRON_BOLT = EntityType.register(
        "${Constants.MOD_ID}:iron_bolt",
        EntityType.Builder.create(EntityIronBolt::class.java) { EntityIronBolt(it) }
            .tracker(50, 1, true)
    )
    val SILVER_BOLT = EntityType.register(
        "${Constants.MOD_ID}:silver_bolt",
        EntityType.Builder.create(EntitySilverBolt::class.java) { EntitySilverBolt(it) }
            .tracker(50, 1, true)
    )
    val IGNEOUS_BOLT = EntityType.register(
        "${Constants.MOD_ID}:igneous_bolt",
        EntityType.Builder.create(EntityIgneousBolt::class.java) { EntityIgneousBolt(it) }
            .tracker(50, 1, true)
    )
    val STAR_METAL_BOLT = EntityType.register(
        "${Constants.MOD_ID}:star_metal_bolt",
        EntityType.Builder.create(EntityStarMetalBolt::class.java) { EntityStarMetalBolt(it) }
            .tracker(50, 1, true)
    )

    // An array containing a list of entities that AOTD adds
    var ENTITY_LIST = arrayOf(
        ENCHANTED_SKELETON,
        WEREWOLF,
        GHASTLY_ENARIA,
        SPLINTER_DRONE,
        SPLINTER_DRONE_PROJECTILE,
        ENARIA,
        ENCHANTED_FROG,
        SPELL_PROJECTILE,
        WOODEN_BOLT,
        IRON_BOLT,
        SILVER_BOLT,
        IGNEOUS_BOLT,
        STAR_METAL_BOLT
    )
}